packager/media/formats/webm/multi_segment_segmenter.cc

```diff
@@ -32,10 +32,14 @@ Status MultiSegmentSegmenter::FinalizeSegment(uint64_t start_timestamp,
     return Status(error::FILE_FAILURE, "Error finalizing segment.");
 
   if (!is_subsegment) {
-    const std::string segment_name = writer_->file()->file_name();
-    // Close the file, which also does flushing, to make sure the file is
-    // written before manifest is updated.
-    RETURN_IF_ERROR(writer_->Close());
+    std::string segment_name =
+        GetSegmentName(options().segment_template, start_timestamp,
+                       num_segment_, options().bandwidth);
+
+    if (!writer_->WriteToFile(segment_name))
+      return Status(error::FILE_FAILURE, "Unable to write file.");
```

Comment:
A simpler approach without the need to modify MkvWriter is to use `File::Copy`. Also, the file needs to be deleted otherwise the memory will not be de-allocated, which could result in memory leak.

// Close the file, which also does flushing, to make sure the file is
// written before manifest is updated.
RETURN_IF_ERROR(writer_->Close());

if (!File::Copy(temp_file_name_.c_str(), segment_name.c_str())) {
  ...
}
if (!File::Delete(temp_file_name_.c_str()) {
  ...
}